crates/astria-sequencer/src/app/vote_extension.rs

```diff
@@ -620,46 +626,85 @@ fn aggregate_oracle_votes(
         }
     }
 
-    let mut prices = HashMap::new();
-    for (currency_pair, mut price_list) in currency_pair_to_price_list {
-        price_list.sort_unstable();
-        let midpoint = price_list
-            .len()
-            .checked_div(2)
-            .expect("has a result because RHS is not 0");
-        let median_price = if price_list.len() % 2 == 0 {
-            'median_from_even: {
-                let Some(left) = price_list.get(midpoint) else {
-                    break 'median_from_even None;
-                };
-                let Some(right_idx) = midpoint.checked_add(1) else {
-                    break 'median_from_even None;
-                };
-                let Some(right) = price_list.get(right_idx).copied() else {
-                    break 'median_from_even None;
-                };
-                left.checked_add(right).and_then(|sum| sum.checked_div(2))
-            }
-        } else {
-            price_list.get(midpoint).copied()
-        }
-        .unwrap_or_else(|| Price::new(0));
-        prices.insert(currency_pair.clone(), median_price);
+    currency_pair_to_price_list
+        .into_iter()
+        .map(|(currency_pair, price_list)| (currency_pair, median(price_list)))
+}
+
+fn median(mut price_list: Vec<Price>) -> Price {
+    price_list.sort_unstable();
+    let midpoint = price_list
+        .len()
+        .checked_div(2)
+        .expect("can't fail as divisor is not zero");
+    if price_list.len() % 2 == 1 {
+        return price_list
+            .get(midpoint)
+            .copied()
+            .expect("`midpoint` is a valid index");
     }
 
-    prices
+    let Some(lower_index) = midpoint.checked_sub(1) else {
+        // We can only get here if `price_list` is empty; just return 0.
+        return Price::new(0);
```

Comment:
the price list should never be empty if constructed correctly in the caller, arguably this should error/panic as silently returning 0 could be bad

Reply:
I agree as currently called, this line should never get hit.  I'll change this to return an error, but in `apply_prices_from_vote_extensions` where we iterate over all the medians, I'll just skip any medians which errored.  That way, a bug here won't halt block production.